compiler/rustc_session/src/config.rs

```diff
@@ -1418,102 +1440,83 @@ impl CheckCfg {
+        //
+        // NOTE: This should be kept in sync with `default_configuration`.
+        // Though symbols inserted conditionally in `default_configuration` are
+        // inserted unconditionally here.
 
-        let atomic_values = &[
-            sym::ptr,
-            sym::integer(8usize),
-            sym::integer(16usize),
-            sym::integer(32usize),
-            sym::integer(64usize),
-            sym::integer(128usize),
-        ];
+        ins!(sym::debug_assertions, no_values);
+
+        // njn: not in default_configuration!
+        ins!(sym::doc, no_values);
+        // njn: not in default_configuration!
+        ins!(sym::doctest, no_values);
+
+        // njn: not in default_configuration!
+        ins!(sym::miri, no_values);
+
+        ins!(sym::overflow_checks, no_values);
+
+        ins!(sym::panic, empty_values).extend(&PanicStrategy::all());
+
+        ins!(sym::proc_macro, no_values);
+
+        ins!(sym::relocation_model, empty_values).extend(RelocModel::all());
 
         let sanitize_values = SanitizerSet::all()
             .into_iter()
             .map(|sanitizer| Symbol::intern(sanitizer.as_str().unwrap()));
+        ins!(sym::sanitize, empty_values).extend(sanitize_values);
 
-        let relocation_model_values = RelocModel::all();
-
-        // Unknown possible values:
-        //  - `target_feature`
-        for name in [sym::target_feature] {
-            self.expecteds.entry(name).or_insert(ExpectedValues::Any);
-        }
-
-        // No-values
-        for name in [
-            sym::doc,
-            sym::miri,
-            sym::unix,
-            sym::test,
-            sym::doctest,
-            sym::windows,
-            sym::proc_macro,
-            sym::debug_assertions,
-            sym::overflow_checks,
-            sym::target_thread_local,
-        ] {
-            self.expecteds.entry(name).or_insert_with(no_values);
-        }
-
-        // Pre-defined values
-        self.expecteds.entry(sym::panic).or_insert_with(empty_values).extend(panic_values);
-        self.expecteds.entry(sym::sanitize).or_insert_with(empty_values).extend(sanitize_values);
-        self.expecteds
-            .entry(sym::target_has_atomic)
-            .or_insert_with(no_values)
-            .extend(atomic_values);
-        self.expecteds
-            .entry(sym::target_has_atomic_load_store)
-            .or_insert_with(no_values)
-            .extend(atomic_values);
-        self.expecteds
-            .entry(sym::target_has_atomic_equal_alignment)
-            .or_insert_with(no_values)
-            .extend(atomic_values);
-        self.expecteds
-            .entry(sym::relocation_model)
-            .or_insert_with(empty_values)
-            .extend(relocation_model_values);
-
-        // Target specific values
+        // njn: sym::sanitizer_cfi_normalize_integers missing!
+        // njn: sym::sanitizer_cfi_normalize_pointers missing!
```

Comment:
These should be added I think.

Reply:
They're expected to be used by users with the `cfi_encoding` attribute. For example, they're currently used by the [cfi_types crate](https://crates.io/crates/cfi-types).

Reply:
Right, in that case let's define them like this:
```suggestion
        ins!(sym::sanitizer_cfi_normalize_integers, no_values););
        ins!(sym::sanitizer_cfi_normalize_pointers, no_values););
```
------
Btw, I don't see those cfgs in [`GATED_CFGS`](https://github.com/rust-lang/rust/blob/8a7b2035f816f6d27003a9326d6bd3a3d739fcc3/compiler/rustc_feature/src/builtin_attrs.rs#L25-L39), the list of unstable cfgs. I think they should be added there since they are unstable, right?